lib/world.rb

```diff
@@ -8,13 +8,28 @@ def initialize(cells)
 
   def evolve
     next_generation = []
-      @cells.each {|cell|
-      neighbours_count = count_live_neighbours(cell.position)
-        if ideal_number_of_neighbours(neighbours_count)
-          next_generation.push(cell)
-        end
-      }
-      @cells = next_generation
+    @cells.each do |cell|
+      if survive?(cell.position)
+        next_generation.push(cell)
+      end
+    end
+    positions_of_potential_resuscitating_cells.each do |position|
+      if resuscitate?(position)
+        new_cell = new_instance_of_cell(position)
```

Comment:
I think it would be better not to wrap `Cell.new(position) into a method. It's a bit unnecessary and just makes the code a bit more difficult to read. It's obvious was `Cell.new` does, so no need to give it a name by wrapping it in a method.